[FEAT] 상세 조회 서버 통신 (#104)

seuriseuljjeok opened the pull request:
## 🔥*Pull requests*

⛳️ **작업한 브랜치**
- feature/#104

👷 **작업한 내용**
<!-- 작업한 내용을 적어주세요. -->
- 상세 조회 서버 통신
## 🚨 참고 사항
<!-- 참고할 사항이 있다면 적어주세요. -->
- 데이터를 넘길 때, 구조체에서 필요한 값이 무엇인지 우선 확인하고 넘기기!
- 그리고 delegate 빼는거 잘 빼기
- 네트워크 호출하는 시점 체크잘하기

### 1.  상세 조회
- 완료/미완료 버튼 클릭시 myToDoData에서 해당하는 투두를 지워주고 서버통신을 해줍니다. 그리고 다시 myToDoData를 리로드 해줍니다.
```swift
//상세 조회 서버 통신
override func viewWillAppear(_ animated: Bool) {
    getDetailToDoData(todoId: self.todoId)
}

// 서버 통신 후 데이터 저장 및 업데이트
var data: GetDetailToDoResponseStuct? {
      didSet {
          guard let data else {return}
          todoTextfield.text = data.title
          deadlineTextfieldLabel.text = data.endDate
          manager = data.allocators
          if data.secret == true {
              self.manager[0].name = "혼자할일"
              self.manager.append(Allocators.EmptyData)
          }
          memoTextView.text = data.memo
          ...
          self.todoManagerCollectionView.reloadData()
      }
  }
```

## 📟 관련 이슈
- Resolved: #104

seuriseuljjeok committed:
[Fix] 투두 레이아웃 수정 (#104)

seuriseuljjeok committed:
[Fix] 아워 & 마이 투두 레이아웃 수정 (#104)

seuriseuljjeok committed:
[Feat] MytoDo 에러 해결 (#104)

0seoYun reviewed (approved):
서버 여러 개라 힘들었을텐데,, 수고 많으셨습니다! 짱희슬🌷

sjk4618 reviewed (approved):
희슬님 최고십니다!!!!

LGTM!

seuriseuljjeok committed:
[Fix] Conflict 해결(#104)

# Conflicts:
#	Going-iOS.xcodeproj/project.pbxproj
#	Going-iOS/Application/SceneDelegate.swift
#	Going-iOS/Scene/OurToDo/ViewControllers/OurToDoViewController.swift
